### PR TITLE
DOCS-3423: Update training script to parse labels based on feedback from etai/tahiya

### PR DESCRIPTION
**Pull request by sguequierre**
* parsing labels and adding validation for num_epochs (used later if they decide to implement a fit function)
* removing google cloud imports bc already gets imported 
Manual testing notes:
* did test and successfully submitted training job with the labels which succeeded with cli (not with UI bc you have to put labels in the arg, put notes for that on docs and made alternate template with the corrections I found but without parsing labels)

### CODE REVIEW

File: docs/data-ai/ai/train.md [hunk 1]
```diff
@@ -250,37 +261,43 @@ def save_model(
         model_dir: output directory for model artifacts
         model_name: name of saved model
     """
-    file_type = ""
-
-    # Save the model to the output directory.
+    file_type = "tflite"  # Add proper file type
```
Comment (sguequierre) [inline docs/data-ai/ai/train.md:264]:
```suggestion
    # Save the model to the output directory
    file_type = "tflite"  # Add proper file type
```

File: docs/data-ai/ai/train.md [hunk 2]
```diff
@@ -210,8 +222,7 @@ def parse_filenames_and_bboxes_from_json(
 
 # Build the model
 def build_and_compile_model(
-    labels: ty.List[str], model_type: str, input_shape: ty.Tuple[int, int, int]
-) -> Model:
+    labels: ty.List[str], model_type: str, input_shape: ty.Tuple[int, int, int]) -> Model:
```
Comment (sguequierre) [inline docs/data-ai/ai/train.md:225]:
```suggestion
    labels: ty.List[str], model_type: str, input_shape: ty.Tuple[int, int, int]
) -> Model:
```

File: docs/data-ai/ai/train.md [hunk 3]
```diff
@@ -108,29 +108,49 @@ unknown_label = "UNKNOWN"
...
+
+    labels = [label.strip() for label in args.labels.strip("'").split()]
+    return args.data_json, args.model_dir, args.num_epochs, labels
 
 
 # This is used for parsing the dataset file (produced and stored in Viam),
 # parse it to get the label annotations
 # Used for training classifiction models
+
+
```
Comment (sguequierre) [inline docs/data-ai/ai/train.md:153]: flake8 parser was complaining about this

File: docs/data-ai/ai/train.md [hunk 4]
```diff
@@ -250,37 +270,45 @@ def save_model(
...
     # Build and compile model on data
-    model = build_and_compile_model()
+    model = build_and_compile_model(image_labels, model_type, IMG_SIZE + (3,))
 
     # Save labels.txt file
     save_labels(LABELS + [unknown_label], MODEL_DIR)
     # Convert the model to tflite
     save_model(
-        model, MODEL_DIR, "classification_model", IMG_SIZE + (3,)
+        model, MODEL_DIR, "classification_model"
```
Comment (sguequierre) [inline docs/data-ai/ai/train.md:311]: correction, was never being used

**Review by vpandiarajan20 — APPROVED** [review]
LGTM! Make sure to try out the script on Viam and add some details about manual testing in the PR comments, then it'll be good to merge

File: docs/data-ai/ai/train.md [hunk 5]
```diff
@@ -108,23 +108,29 @@ unknown_label = "UNKNOWN"
...
 
+DEFAULT_EPOCHS = 200
 
 # This parses the required args for the training script.
 # The model_dir variable will contain the output directory where
 # the ML model that this script creates should be stored.
 # The data_json variable will contain the metadata for the dataset
 # that you should use to train the model.
+
+
```
Comment (sguequierre) [inline docs/data-ai/ai/train.md:119]: know this is awk but python markdown parser was complaining about this specifically

File: docs/data-ai/ai/train.md [hunk 6]
```diff
@@ -300,6 +541,10 @@ The script you are creating must take the following command line inputs:
 - `dataset_file`: a file containing the data and metadata for the training job
 - `model_output_directory`: the location where the produced model artifacts are saved to
 
+If you used the version with labels parsed, it will also take the following command line inputs:
```
Comment (npentrel) [inline docs/data-ai/ai/train.md:544]:
```suggestion
If you used the version with parsed labels, it will also take the following command line inputs:
```

File: docs/data-ai/ai/train.md [hunk 7]
```diff
@@ -547,6 +792,11 @@ In the Viam app, navigate to your list of [**DATASETS**](https://app.viam.com/da
 
 Click **Train model** and select **Train on a custom training script**, then follow the prompts.
 
+{{% alert title="Tip" color="tip" %}}
+If you used the version of <file>training.py</file> with labels parsed, your training job will fail with the error `ERROR training.py: error: the following arguments are required: --labels`.
```
Comment (npentrel) [inline docs/data-ai/ai/train.md:796]:
```suggestion
If you used the version of <file>training.py</file> with parsed labels, your training job will fail with the error `ERROR training.py: error: the following arguments are required: --labels`.
```

File: docs/data-ai/ai/train.md [hunk 8]
```diff
@@ -547,6 +792,11 @@ In the Viam app, navigate to your list of [**DATASETS**](https://app.viam.com/da
...
+Use the CLI and enter labels.
```
Comment (npentrel) [inline docs/data-ai/ai/train.md:797]:
```suggestion
To use labels, you must use the CLI.
```

File: docs/data-ai/ai/train.md [hunk 9]
```diff
@@ -90,15 +88,17 @@ If you haven't already, create a folder called <file>model</file> and create an
 
 <p><strong>4. Add <code>training.py</code> code</strong></p>
 
-<p>Copy this template into <file>training.py</file>:</p>
+<p>Copy one of the following templates into <file>training.py</file>, depending on whether or not you wish to parse labels:</p>
```
Comment (npentrel) [inline docs/data-ai/ai/train.md:91]: Can you rephrase this to have the condition based on which the reader should decide be clearer? Why would a user need to use the labels flag or not?

**Review by npentrel — CHANGES_REQUESTED** [review]
see comment about conditional

File: docs/data-ai/ai/train.md (hunk 9 above)
Comment (sguequierre) [inline docs/data-ai/ai/train.md:91]:
```suggestion
<p>Copy one of the following templates into <file>training.py</file>, depending on whether or not you wish to parse labels instead of having to hard code the labels in the training script and re-upload for each set of labels, allowing you to more easily reuse training scripts:</p>
```

File: docs/data-ai/ai/train.md [hunk 10]
```diff
@@ -273,14 +284,244 @@ if __name__ == "__main__":
...
     # Convert the model to tflite
     save_model(
-        model, MODEL_DIR, "classification_model", IMG_SIZE + (3,)
+        model, MODEL_DIR, "classification_model"
+    )
+```
+
+{{% /expand %}}
+
+{{% expand "Click to see the template with labels parsed" %}}
```
Comment (sguequierre) [inline docs/data-ai/ai/train.md:306]:
```suggestion
{{% expand "Click to see the template with parsed labels" %}}
```

File: docs/data-ai/ai/train.md [hunk 11]
```diff
@@ -90,15 +88,17 @@ If you haven't already, create a folder called <file>model</file> and create an
 
 <p><strong>4. Add <code>training.py</code> code</strong></p>
 
-<p>Copy this template into <file>training.py</file>:</p>
+<p>Copy one of the following templates into <file>training.py</file>, depending on whether or not you wish to parse labels instead of having to hard code the labels in the training script and re-upload for each set of labels, allowing you to more easily reuse training scripts:</p>
```
Comment (sguequierre) [inline docs/data-ai/ai/train.md:91]:
```suggestion
<p>Copy one of the following templates into <file>training.py</file>, depending on whether or not you wish to parse labels instead of having to hard code the labels in the training script and re-upload for each set of labels, which allows you to more easily reuse training scripts:</p>
```

File: docs/data-ai/ai/train.md [hunk 12]
```diff
@@ -90,15 +88,17 @@ If you haven't already, create a folder called <file>model</file> and create an
 
 <p><strong>4. Add <code>training.py</code> code</strong></p>
 
-<p>Copy this template into <file>training.py</file>:</p>
+<p>Copy one of the following templates into <file>training.py</file>, depending on whether or not you wish to parse labels instead of having to hard code the labels in the training script and re-upload for each set of labels, which allows you to more easily reuse training scripts:</p>
```
Comment (npentrel) [inline docs/data-ai/ai/train.md:91]: This is now a very long sentence and it makes it hard to understand where labels come from.

```suggestion
<p>You can set up your training script to use a hard coded set of labels or allow users to pass in a set of labels when using the training script. Allowing users to pass in labels when using training scripts makes your training script more flexible for reuse.</p>
<p>Copy one of the following templates into <file>training.py</file>, depending on how you want to handle labels:</p>
```

Do you see the difference?

File: docs/data-ai/ai/train.md [hunk 13]
```diff
@@ -300,6 +541,10 @@ The script you are creating must take the following command line inputs:
 - `dataset_file`: a file containing the data and metadata for the training job
 - `model_output_directory`: the location where the produced model artifacts are saved to
 
+If you used the version with parsed labels, it will also take the following command line inputs:
```
Comment (npentrel) [inline docs/data-ai/ai/train.md:544]:
```suggestion
If you used the training script template that allows users to pass in lables, it will also take the following command line inputs:
```

File: docs/data-ai/ai/train.md [hunk 14]
```diff
@@ -547,6 +792,11 @@ In the Viam app, navigate to your list of [**DATASETS**](https://app.viam.com/da
 
 Click **Train model** and select **Train on a custom training script**, then follow the prompts.
 
+{{% alert title="Tip" color="tip" %}}
+If you used the version of <file>training.py</file> with parsed labels, your training job will fail with the error `ERROR training.py: error: the following arguments are required: --labels`.
```
Comment (npentrel) [inline docs/data-ai/ai/train.md:796]:
```suggestion
If you used the version of <file>training.py</file> that allows users to pass in lables, your training job will fail with the error `ERROR training.py: error: the following arguments are required: --labels`.
```

File: docs/data-ai/ai/train.md (hunk 12 above)
Comment (sguequierre) [reply docs/data-ai/ai/train.md:91]: Yes, I do see the difference. Will keep that in mind. Thank you!

**Review by npentrel — APPROVED** [review]
